### PR TITLE
Structural and Performance improvements

### PR DESCRIPTION
**Pull request by balasankarc**
### General
- Use more sensible namespace: Follow `libindic.<module>.<language>` structure
- Migrate to testtools for testing
- Use pbr for packaging
- Add CI+Coverage for analysis and Tox for local automation.
### Malayalam
- Minimize false positives by using a root word corpus
- Handle multilevel inflections using iterative suffix stripping
- Add more tests

### CODE REVIEW

**Comment by balasankarc** [comment]
Go home coveralls, you are drunk. You got it in the wrong order. :smile:

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 travis:
 	python setup.py test --coverage \
-		--coverage-package-name=stemmer
-	flake8 --max-complexity 10 libindic/stemmer
+		--coverage-package-name=libindic.stemmer
+	flake8 --max-complexity 15 libindic/stemmer
```
Comment (copyninja) [inline Makefile]: Why 15?. The value 10 is considered to be normal

**Comment by copyninja** [comment]
One general comment at many places comments doesn't make much sense like Travis Autopep8 and many places I note that commit message and commit content doesn't go together. Eg. Polish travis is commit message, commit contains changes to files under doc/ and many unrelated files.

This is not acceptable do atomic commits, i.e. commit only related files so that at any time we can go back and revert things as needed. For now I'm okay with this but if possible try to rebase and give proper commit messages than single word.

File: Makefile (hunk 1 above)
Comment (balasankarc) [inline Makefile]: Ok. Malayalam.stem method had complexity of 10 and that was failing flake8 with max-complexity set to 10. Let me see if I can reduce that complexity and maintain max-complexity at 10. If it is not reducible, what shall I do?

BTW, existing Punjabi stemmer seems to have a complexity of 12, which will fail flake8 test. I don't intend to touch the Punjabi/Hindi stemmer for GSoC, as I am not familiar with their grammatical properties. Can I just delete those languages for now?

**Comment by balasankarc** [comment]
Ok. I will keep that in mind and make atomic commits in future. Sorry for the trouble. :)

File: Makefile (hunk 1 above)
Comment (copyninja) [inline Makefile]: Just to give brief on complexity, it increases as you nest too much eg. conditionals. The way to reduce it refactor code such a way you don't nest too much. I've not checked the code so I can't give you exact reason but that is the idea.

I think it was added by @arushidogra lets see if she can help or not I will take a look later.

**Comment by copyninja** [comment]
Good try to reduce complexity of your code and I shall merge the PR.

File: Makefile (hunk 1 above)
Comment (balasankarc) [inline Makefile]: Yeah. I've brought the complexity of Malayalam stemmer under 10.

Comment (copyninja) [inline Makefile]: Did you forget to push?. I didn't see the related comit yet

**Comment by copyninja** [comment]
OK I'm merging the request now

**Comment by balasankarc** [comment]
@copyninja Thanks.

PS [OT]: BTW, is your name "Vasudev" or "Vasudeva" ? I see both of them intermixed in commits/mail notifications. :)

**Comment by copyninja** [comment]
Err.. I did a mistake to use email id and name which was not supposed to be there. I've nuked the history sorry about it. Please pull carefully.

And phonetically its Vaasudeva but English is not phonetic ;-)

**Comment by copyninja** [comment]
@balasankarc Please pull before you start working further!. I've renuked the history.

**Comment by balasankarc** [comment]
@copyninja Ok.

**Pull request merged**